network/src/topology.rs

```diff
@@ -218,8 +218,9 @@ impl KnownNetwork {
         self.nodes.read().clone()
     }
 
-    /// Returns a map of the potential forks.
-    pub fn potential_forks(&self) -> HashMap<u32, Vec<SocketAddr>> {
+    #[allow(clippy::type_complexity)]
+    /// Returns the canon tip height and members, and a map of the potential forks.
+    pub fn potential_forks(&self) -> (Option<(u32, Vec<SocketAddr>)>, HashMap<u32, Vec<SocketAddr>>) {
```

Comment:
maybe we can reuse the `PotentialFork` (or `NodeCluster`, as suggested later on) here for a simpler return type?

Reply:
I wanted to avoid a dependency on the RPC types but I can certainly wrap these in a new type?

Reply:
maybe the RPC type could instead reside here?

Reply:
Fine by me 👍